File: CreditCard.js
const UniqueItem = require("./UniqueItem");
const UtilsMethod = require("./UtilsMethod");

const prompt = require("prompt-sync")();
class CreditCard extends UniqueItem {
  constructor(bank, cardType, expirationDate, ownerName) {
    super("CreditCard");
    this.bank = bank;
    this.cardType = cardType;
    this.expirationDate = expirationDate;
    this.ownerName = ownerName;
  }

  static createCreditCard() {
    const bank = prompt("Entrez la banque de la carte de crédit : ");
    const cardType = prompt("Entrez le type de carte de crédit : ");
    console.log("Entrez la date d'expiration : ");
    const expirationDate = UtilsMethod.askDate();
    const ownerName = prompt("Entrez le nom du propriétaire : ");

    // Vous pouvez ajouter une logique de validation pour la date d'expiration si nécessaire

    return new CreditCard(bank, cardType, expirationDate, ownerName);
  }

  displayInfo() {
    console.log("Informations de la carte de crédit :");
    console.log(`Banque : ${this.bank}`);
    console.log(`Type de carte : ${this.cardType}`);
    console.log(`Date d'expiration : ${this.expirationDate}`);
    console.log(`Propriétaire : ${this.ownerName}`);
  }

  toString() {
    console.log("Informations de la carte de crédit :");
    console.log(`Banque : ${this.bank}`);
    console.log(`Type de carte : ${this.cardType}`);
    console.log(`Date d'expiration : ${this.expirationDate}`);
    console.log(`Propriétaire : ${this.ownerName}`);
    return "";
  }
}

module.exports = CreditCard;

File: CreditCard.py
from UniqueItem import UniqueItem
from UtilsMethod import UtilsMethod


class CreditCard(UniqueItem):
    def __init__(self, bank, card_type, expiration_date, owner_name, name):
        super().__init__(name)
        self.bank = bank
        self.card_type = card_type
        self.expiration_date = expiration_date
        self.owner_name = owner_name

    @classmethod
    def from_user_input():
        print("Enter bank: ")
        bank = input()
        print("Enter card type: ")
        card_type = input()
        print("Enter expiration date: ")
        expiration_date = UtilsMethod.ask_date(input)
        print("Enter owner name: ")
        owner_name = input()

        return CreditCard(bank, card_type, expiration_date, owner_name, "CreditCard")

    def __str__(self):
        return (
            f"CreditCard{{bank='{self.bank}', cardType='{self.card_type}', "
            f"expirationDate={self.expiration_date}, ownerName='{self.owner_name}'}}"
        )
